main.go

```diff
@@ -106,6 +107,36 @@ func truncateDate(date time.Time, granularity string) time.Time {
 	}
 }
 
+// Calculates whether or not input data (s3) is more stale than target data (Redshift)
+// Expects:
+// - inputDataDate corresponds to the s3 data timestamp of the job
+// - targetDataDate is the maximum timestamp of the DB table
+// - granularity indicating how often data snapshots are recorded in the target
+func isInputDataStale(inputDataDate time.Time, targetDataDate *time.Time, granularity string, isTargetDataPT bool) bool {
+	// If target table has no data, then input data is fresh by default
+	if targetDataDate == nil {
+		return false
+	}
+
+	// If the target data is actually in PT, then we need to compare accordingly
+	if isTargetDataPT {
+		ptLoc, err := time.LoadLocation("America/Los_Angeles")
+		fatalIfErr(err, "isInputDataStale was unable to load PT timezone")
+
+		_, ptOffsetSec := targetDataDate.In(ptLoc).Zone()
+		ptOffsetDuration, err := time.ParseDuration(fmt.Sprintf("%vs", -1*ptOffsetSec))
+		fatalIfErr(err, "isInputDataStale was unable to parse offset duration")
+
+		targetDataDateInUTC := targetDataDate.Add(ptOffsetDuration)
+		targetDataDate = &targetDataDateInUTC
+	}
+
+	// We truncate the timestamps to make the comparison at the correct granularity
+	// i.e. input data lagging by two hours is considered stale when granularity is hourly,
+	// but it can still be considered fresh when the granularity is daily.
+	return truncateDate(*targetDataDate, granularity).After(truncateDate(inputDataDate, granularity))
```

Comment:
nit: I think it is possible to replace truncateDate with startEndFromGranularity. It might be a bit out of scope, but could also simplify understanding of the file.